openpype/hosts/maya/plugins/publish/validate_resolution.py

```diff
@@ -0,0 +1,54 @@
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from maya import cmds
+from openpype.hosts.maya.api.lib import reset_scene_resolution
+
+
+class ValidateResolution(pyblish.api.InstancePlugin,
+                         OptionalPyblishPluginMixin):
+    """Validate the resolution setting aligned with DB"""
+
+    order = pyblish.api.ValidatorOrder - 0.01
+    families = ["renderlayer"]
+    hosts = ["maya"]
+    label = "Validate Resolution"
+    optional = True
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+        width, height = self.get_db_resolution(instance)
+        current_width = cmds.getAttr("defaultResolution.width")
+        current_height = cmds.getAttr("defaultResolution.height")
+        if current_width != width and current_height != height:
+            raise PublishValidationError("Resolution Setting "
+                                         "not matching resolution "
+                                         "set on asset or shot.")
+        if current_width != width:
+            raise PublishValidationError("Width in Resolution Setting "
+                                         "not matching resolution set "
+                                         "on asset or shot.")
+
+        if current_height != height:
+            raise PublishValidationError("Height in Resolution Setting "
+                                         "not matching resolution set "
+                                         "on asset or shot.")
+
+    def get_db_resolution(self, instance):
+        asset_doc = instance.data["assetEntity"]
+        project_doc = instance.context.data["projectEntity"]
+        for data in [asset_doc["data"], project_doc["data"]]:
+            if "resolutionWidth" in data and "resolutionHeight" in data:
+                width = data["resolutionWidth"]
+                height = data["resolutionHeight"]
+                return int(width), int(height)
+
+        # Defaults if not found in asset document or project document
+        return 1920, 1080
+
+    @classmethod
+    def repair(cls, instance):
+        return reset_scene_resolution()
```

Comment:
This being renderlayers complicates this a bit, because technically there _can_ be renderlayer overrides on a single layer. I suspect what repair should be doing is triggering the repair onto the layer to the best of its capabilities, so that _if_ there's a renderlayer override that it sets it there.

Also here not that render resolution _can_ be a different attribute than just `defaultResolution.width` e.g. like the V-Ray case I commented above.

Reply:
By the way, **note** that `reset_scene_resolution` **also** sets the Pixel Aspect Ratio which this validator does **not** validate/report about which might be confusing. As such, we might want to add that as a validation in here too so it validates resolution + pixel aspect?

Reply:
We can have two validators for the resolution, one is for scene resolution and the other is for render resolution.

Reply:
Not sure whether that's necessary and it'd be a bit confusing because for most renderers scene resolution == render resolution.